lighthouse-core/gather/gather-runner.js

```diff
@@ -420,6 +422,16 @@ class GatherRunner {
         // Save devtoolsLog, but networkRecords are discarded and not added onto artifacts.
         baseArtifacts.devtoolsLogs[passConfig.passName] = passData.devtoolsLog;
 
+        const userAgentEntry = passData.devtoolsLog.find(entry =>
+          entry.method === 'Network.requestWillBeSent' &&
+          !!entry.params.request.headers['User-Agent']
+        );
+
+        if (userAgentEntry && !baseArtifacts.NetworkUserAgent) {
+          // @ts-ignore - guaranteed to exist by the find above
+          baseArtifacts.NetworkUserAgent = userAgentEntry.params.request.headers['User-Agent'];
+        }
```

Comment:
could also do `baseArtifacts.NetworkUserAgent = await driver.getUserAgent()` after `GatherRunner.setupDriver()` is called. It's shorter, but is it better to just look at the actual network traffic?

Reply:
Given the number of ways that have popped up in issues as the way folks are trying to set the user agent string, I thought it was best to grab it directly from network records instead of assume it's done being manipulated after setupDriver

Reply:
SG. I guess `HostUserAgent` might end up weird too if people are overriding things